stan/math/prim/meta/is_var_and_matrix_types.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef STAN_MATH_PRIM_META_IS_VAR_AND_MATRIX_TYPES_HPP
+#define STAN_MATH_PRIM_META_IS_VAR_AND_MATRIX_TYPES_HPP
+
+#include <stan/math/prim/meta/disjunction.hpp>
+#include <stan/math/prim/meta/is_var.hpp>
+#include <stan/math/prim/meta/is_matrix.hpp>
+#include <stan/math/prim/meta/require_helpers.hpp>
+#include <stan/math/prim/meta/return_type.hpp>
+
+namespace stan {
+
+/** \ingroup type_trait
+ * Extends std::true_type when instantiated with one type that has a var
+ * `scalar_type` and another that is a matrix. Extends std::false_type
```

Comment:
Might be more accurate to say that at least one type has `scalar_type` var and at least one type is a matrix, since the current wording implies that the check only returns true when there is a `var` type and another `matrix` type, not if there's a `matrix<var>` type